container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/docker_kurtosis_backend_api_container_functions.go

```
@@ -551,6 +561,23 @@ func getApiContainerObjectFromContainerInfo(
 	return result, nil
 }
 
+func getApiContainerObjectFromContainerInfoWithBridgeNetworkIpAddress(
+	apicObj *api_container.APIContainer,
+	bridgeNetworkIpAddress string,
+) (*api_container.APIContainer, error) {
+	result := api_container.NewAPIContainer(
+		apicObj.GetEnclaveID(),
+		apicObj.GetStatus(),
+		apicObj.GetPrivateIPAddress(),
+		apicObj.GetPrivateGRPCPort(),
+		apicObj.GetPublicIPAddress(),
+		apicObj.GetPublicGRPCPort(),
+		net.ParseIP(bridgeNetworkIpAddress),
```

ParseIP has a silent error condition:
```
		bridgeNetworkIpAddr = net.ParseIP(bridgeNetworkIpAddrStr)
		if bridgeNetworkIpAddr == nil {
			return nil, stacktrace.NewError("Couldn't parse bridge network IP address string '%v' to an IP", bridgeNetworkIpAddrStr)
		}
```

---

okay for this to be nil here, as the api container can be in states where its not connected to the bridge network. added a comment clarifying in the api container object